File: 7-ReverseInteger/7-ReverseInteger.java
// Last updated: 8/1/2025, 10:42:59 PM
class Solution {
    public int reverse(int x) {
        String s = String.valueOf(x);
        char[] arr = s.toCharArray();
        
        if (arr[0] != '-') {
            int i = 0;
            int j = arr.length - 1;
            while (i < j) {
                char temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;
                j--;
            }
        } else {
            int i = 1;
            int j = arr.length - 1;
            while (i < j) {
                char temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
                i++;
                j--;
            }
        }

        String s2 = new String(arr);
        try {
            int ans = Integer.parseInt(s2);
            return ans;
        } catch (NumberFormatException e) {
            return 0; // Handle overflow case
        }
    }
}

File: 7-ReverseInteger/7-ReverseInteger.c
// Last updated: 8/1/2025, 10:43:05 PM
// int reverse(int x){

// #include <limits.h>

int reverse(int x) {
    int result = 0;

    while (x != 0) {
        int digit = x % 10;

        // Check for overflow
        if (result > INT_MAX / 10 || (result == INT_MAX / 10 && digit > 7)) return 0;
        if (result < INT_MIN / 10 || (result == INT_MIN / 10 && digit < -8)) return 0;

        result = result * 10 + digit;
        x /= 10;
    }

    return result;
}
